### PR TITLE
Make the _getInstance() instantiator a util for the whole SDK

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-exp/src/core/util/instantiator.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Unfortunately we can't export these as classes instantiated directly since
+ * the constructor may side effect and therefore can't be proven to be safely
+ * culled. Instead, we export these classes themselves as a lowerCamelCase
+ * constant, and instantiate them under the hood.
+ */
+export interface SingletonInstantiator<T> {
+  new (): T;
+}
+
+const persistenceCache: Map<unknown, unknown> = new Map();
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/util/instantiator.ts:29]: this is no longer persistence specific, maybe call it instanceCache?

File: packages-exp/auth-exp/src/core/util/instantiator.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export function _getInstance<T>(cls: unknown): T {
+  if (persistenceCache.has(cls)) {
+    return persistenceCache.get(cls)! as T;
+  }
+
+  const persistence = new (cls as SingletonInstantiator<T>)();
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/util/instantiator.ts:36]: same here, `const instance`

File: packages-exp/auth-exp/src/core/util/instantiator.ts (hunk 1 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/util/instantiator.ts]: Done

File: packages-exp/auth-exp/src/core/util/instantiator.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/util/instantiator.ts]: Done

**Pull request merged**